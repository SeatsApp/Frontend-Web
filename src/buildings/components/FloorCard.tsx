import {
    Button,
    Card, Typography,
} from "@mui/material";
import { useNavigate } from "react-router-dom";
import { Floor } from "../../shared/types/Floor";

interface FloorCardProps {
    buildingId: number,
    floor: Floor
}

export default function FloorCard({ buildingId, floor }: FloorCardProps) {
    const navigate = useNavigate();

    return (
        <Card style={{ padding: 10, margin: 10 }}>
            <Typography>{floor.name}</Typography>
            <Button
                variant="contained"
                color={'secondary'}
                onClick={() => navigate("/buildings/" + buildingId)}
            >
                Change building
            </Button>
            <Button
                variant="contained"
                color={'secondary'}
                onClick={() => navigate("/buildings/" + buildingId + "/floors/" + floor.id + "/floorplan")}
            >
                Manage floorplan
            </Button>
        </Card>
    )
}