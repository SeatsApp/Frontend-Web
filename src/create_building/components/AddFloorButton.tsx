import { Box, Button } from "@mui/material";
import { Dispatch, SetStateAction } from "react";
import { Building } from "../../shared/types/Building";

interface AddFloorButtonProps {
    building: Building;
    setBuilding: Dispatch<SetStateAction<Building>>,
}

export default function AddFloorButton({ building, setBuilding }: AddFloorButtonProps) {

    const addFloor = () => {
        const floorCount = building.floors.length
        building.floors[floorCount] = {
            id: 0,
            name: "",
            points: [{
                firstPoint: 0,
                secondPoint: 0
            },
            {
                firstPoint: 1600,
                secondPoint: 0
            },
            {
                firstPoint: 1600,
                secondPoint: 1800
            },
            {
                firstPoint: 0,
                secondPoint: 1800
            },
            {
                firstPoint: 0,
                secondPoint: 0
            }],
            seats: []
        }
        setBuilding({ ...building, floors: building.floors })
    }

    return (
        <Box style={{ marginTop: 10, marginBottom: 10 }}>
            <Button variant="outlined" color={'secondary'}
                onClick={addFloor}>
                Add floor</Button>
        </Box>
    )
}