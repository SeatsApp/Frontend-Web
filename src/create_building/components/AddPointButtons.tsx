import { Box, Button } from "@mui/material";
import { Dispatch, SetStateAction } from "react";
import { Building } from "../../shared/types/Building";

interface AddPointButtonProps {
    floorIndex: number
    building: Building;
    setBuilding: Dispatch<SetStateAction<Building>>,
}

export default function AddPointButton({ floorIndex, building, setBuilding }: AddPointButtonProps) {

    const addPoint = (changeFloorIndex: number) => {
        const floor = building.floors[changeFloorIndex]
        const pointCount = floor.points.length
        floor.points[pointCount] = {
            firstPoint: 0,
            secondPoint: 0
        }
        building.floors[changeFloorIndex] = floor
        setBuilding({ ...building, floors: building.floors })
    }

    return (
        <Box style={{ marginTop: 10, marginBottom: 10 }}>
            <Button variant="contained" color={'secondary'}
                onClick={() => addPoint(floorIndex)}>
                Add point</Button>
        </Box>
    )
}