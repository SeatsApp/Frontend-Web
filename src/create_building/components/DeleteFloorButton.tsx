import { IconButton } from "@mui/material";
import { Dispatch, SetStateAction } from "react";
import { Building } from "../../shared/types/Building";
import DeleteIcon from '@mui/icons-material/Delete';

interface DeleteFloorButtonProps {
    building: Building;
    floorIndex: number;
    setBuilding: Dispatch<SetStateAction<Building>>,
}

export default function DeleteFloorButton({ building, floorIndex, setBuilding }: DeleteFloorButtonProps) {
    const deleteFloor = (changeFloorIndex: number) => {
        const newFloors = building.floors.filter((_item, index) => index !== changeFloorIndex);
        setBuilding({ ...building, floors: newFloors })
    }

    return (
        <IconButton style={{ color: "red" }} onClick={() => deleteFloor(floorIndex)}>
            <DeleteIcon />
        </IconButton>
    )
}