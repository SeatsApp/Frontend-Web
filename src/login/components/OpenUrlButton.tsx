import React, {useCallback} from "react";
import {Button, Typography} from "@mui/material";
import LockOpenIcon from '@mui/icons-material/LockOpen';

interface OpenUrlButtonProps {
    url: string;
}

export const OpenURLButton = ({url}: OpenUrlButtonProps) => {

    const handlePress = useCallback(async () => {
        window.location.assign(url);
    }, [url]);

    return <Button sx={{padding: 4}} startIcon={<LockOpenIcon />} color={'info'} variant='contained'
                   onClick={() => handlePress()}><Typography variant='h6'>Connect with your Cronos account.</Typography></Button>;
};