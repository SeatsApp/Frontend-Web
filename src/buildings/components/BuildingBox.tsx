import {
    Box, Typography,
} from "@mui/material";
import { Building } from "../../shared/types/Building";
import { Floor } from "../../shared/types/Floor";
import FloorCard from "./FloorCard";

interface BuildingBoxProps {
    building: Building
}

export default function BuildingBox({ building }: BuildingBoxProps) {
    return (
        <Box style={{ padding: 10, margin: 10 }}>
            <Typography variant='h5'>{building.name}</Typography>
            {building.floors.map((floor: Floor) =>
                <FloorCard key={floor.id} buildingId={building.id} floor={floor} />)}
        </Box>
    )
}