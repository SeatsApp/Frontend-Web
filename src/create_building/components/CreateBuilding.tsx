import {
    Box, Card, TextField, Typography
} from "@mui/material";
import { useParams } from "react-router-dom";
import { Building } from "../../shared/types/Building";
import useReadBuilding from "../hooks/useReadBuilding";
import { useEffect, useState } from "react";
import AddFloorButton from "./AddFloorButton";
import FloorAccordion from "./FloorAccordion";
import { Floor } from "../../shared/types/Floor";
import SaveBuildingButton from "./SaveBuildingButton";

export const CreateBuilding = () => {
    const { buildingId } = useParams()
    const [building, setBuilding] = useState<Building>({
        id: 0,
        name: "",
        floors: []
    })


    if (buildingId !== undefined && buildingId !== "0") {
        // eslint-disable-next-line react-hooks/rules-of-hooks
        const { building: readBuilding, loading } = useReadBuilding(buildingId)

        // eslint-disable-next-line react-hooks/rules-of-hooks
        useEffect(() => {
            setBuilding(readBuilding)
        }, [loading, readBuilding])
    }

    const onChangeBuildingName = (event: any) => {
        setBuilding({ ...building, name: event.target.value })
    }

    return (
        <Box style={{ paddingTop: 20, margin: 2 }}>
            <Card style={{ padding: 20 }}>
                <Typography color={'secondary'} variant='h3'>
                    {buildingId !== "0" ? "Change" : "Create"} building
                </Typography>
                <TextField value={building.name} onChange={onChangeBuildingName}
                    label="Building name" variant="outlined" type="text" />
                <AddFloorButton building={building} setBuilding={setBuilding} />
                {building.floors.map((floor: Floor, floorIndex) => (
                    <FloorAccordion key={floorIndex} floor={floor} floorIndex={floorIndex}
                        building={building} setBuilding={setBuilding} />
                ))}
                <SaveBuildingButton building={building} />
            </Card>
        </Box>
    )
};