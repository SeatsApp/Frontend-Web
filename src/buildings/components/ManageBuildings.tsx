import useReadBuildings from "../hooks/useReadBuildings";
import {
    Box,
    Button,
    Card, Typography,
} from "@mui/material";
import BuildingBox from "./BuildingBox";
import { useNavigate } from "react-router-dom";
import { Building } from "../../shared/types/Building";


export const ManageBuildings = () => {
    const navigate = useNavigate();
    const { buildings } = useReadBuildings()

    return (
        <Box style={{ paddingTop: 20, margin: 2 }}>
            <Card style={{ padding: 20 }}>
                <Box style={{ display: "flex", flexWrap: 'wrap', justifyContent: 'space-between'}}>
                    <Typography color={'secondary'} variant='h3'>Buildings overview</Typography>
                    <Button
                        variant="contained"
                        color={'secondary'}
                        onClick={() => navigate("/buildings/0")}
                    >
                        Create building
                    </Button>
                </Box>
                {buildings.map((building: Building) =>
                    <BuildingBox key={building.id} building={building} />
                )}
            </Card>
        </Box>
    )
};