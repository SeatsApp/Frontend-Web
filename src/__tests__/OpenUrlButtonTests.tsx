import React from "react";
import renderer from "react-test-renderer";
import '@testing-library/jest-dom'
import {OpenURLButton} from "../login/components/OpenUrlButton";
import {fireEvent, render, waitFor} from "@testing-library/react";

beforeEach(() => {
    jest.spyOn(console, 'warn').mockImplementation();
    jest.spyOn(console, 'error').mockImplementation();
});

test("renders correctly", () => {
    const tree = renderer.create(<OpenURLButton url={""} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Test web button press", async () => {
    // @ts-ignore
    delete window.location
    // @ts-ignore
    window.location = { assign: jest.fn() }

    const screen = render(<OpenURLButton url="https://google.be" />);

    fireEvent.click(screen.getByText("Connect with your Cronos account."));

    await waitFor(() => {
        expect(window.location.assign).toBeCalled();
    })
});