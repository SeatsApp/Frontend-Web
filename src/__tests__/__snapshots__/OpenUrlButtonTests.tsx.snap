// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Login
  </p>
</button>
`;
